tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc

```diff
@@ -15932,4 +15933,245 @@ TEST_F(EnergyPlusFixture, VRFTU_FanOnOff_Power)
+    EXPECT_EQ(state->dataErrTracking->NumRecurringErrors, 1);
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors[0].MinValue, -0.25);
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors[0].MaxValue, -0.25);
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors[0].Count, 1);
+    EXPECT_EQ(state->dataErrTracking->RecurringErrors[0].Message,
+              " ** Warning ** AirConditioner:VariableRefrigerantFlow \"\": Cooling EIR Modifier curve (function of PLR) output is negative warning "
+              "continues...");
+
+    // TU's are in heating mode only
+    state->dataHVACVarRefFlow->CoolingLoad(VRFCond) = false;
+    state->dataHVACVarRefFlow->HeatingLoad(VRFCond) = true;
+    state->dataHVACVarRefFlow->LastModeCooling(VRFCond) = false;
+    state->dataHVACVarRefFlow->LastModeHeating(VRFCond) = true;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalCoolLoad(1) = 0.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalCoolLoad(2) = 0.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalCoolLoad(3) = 0.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalCoolLoad(4) = 0.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalCoolLoad(5) = 0.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalHeatLoad(1) = 1000.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalHeatLoad(2) = 1000.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalHeatLoad(3) = 1000.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalHeatLoad(4) = 1000.0;
+    state->dataHVACVarRefFlow->TerminalUnitList(1).TotalHeatLoad(5) = 1000.0;
+
+    CalcVRFCondenser(*state, VRFCond);
+    EXPECT_FALSE(state->dataHVACVarRefFlow->VRF(VRFCond).HRHeatingActive);
+    EXPECT_FALSE(state->dataHVACVarRefFlow->VRF(VRFCond).HRCoolingActive);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).TotalCoolingCapacity, 0.0);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).TUCoolingLoad, 0.0);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).TotalHeatingCapacity, 5000.0);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).TUHeatingLoad, 5000.0);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).VRFCondPLR, 0.25);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).VRFCondRTF, 1.0);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).SUMultiplier, 1.0);
+    EXPECT_FALSE(state->dataHVACVarRefFlow->VRF(VRFCond).ModeChange);
+    EXPECT_FALSE(state->dataHVACVarRefFlow->VRF(VRFCond).HRModeChange);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).ElecCoolingPower, 0.0);
+
+    // This should be reset to zero after the fix
+    EXPECT_NE(state->dataHVACVarRefFlow->VRF(VRFCond).ElecHeatingPower,
+              state->dataHVACVarRefFlow->VRF(VRFCond).RatedHeatingPower * state->dataHVACVarRefFlow->VRF(VRFCond).VRFCondPLR);
+    EXPECT_EQ(state->dataHVACVarRefFlow->VRF(VRFCond).ElecHeatingPower, 0.0);
```

Comment:
Please explain. The VRF just finished simulating. Why is the ElecHeatingPower = 0 if there actually was heating power? I would think heating power would be initialized to 0 in init, after all other reporting was complete and before then next iteration of the model.